src/js/views/patients/worklist/worklist_views.js

```diff
@@ -213,6 +214,20 @@ const ListView = CollectionView.extend({
       });
     });
   },
+  onClickShiftMultiSelect(selectedActionId) {
+    const actionsList = this.children._views;
+    const lastSelectedActionId = this.state.getLastSelected();
+
+    const selectedIndex = actionsList.findIndex(action => action.model.id === selectedActionId);
+    const lastSelectedIndex = actionsList.findIndex(action => action.model.id === lastSelectedActionId);
+
+    const arrayOfActionIdsToSelect = actionsList
+      .slice(Math.min(selectedIndex, lastSelectedIndex), Math.max(selectedIndex, lastSelectedIndex) + 1)
+      .map(action => action.model.id);
+
+    this.state.selectMultiple(arrayOfActionIdsToSelect);
+    this.state.setLastSelectedAction(selectedActionId);
+  },
 });
```

Comment:
Consumes the selection action `id` passed to it from the action child view.

Using the `lastSelectedAction` value, it gets the index position of the current and previous selected actions. And uses those indexes to slice the list of actions to only include the ones between those two indexes.

Then, it creates an array of action `id`s that are passed to the state. `Math.min()` and `Math.max()` allow the user to select both directions in the list (up or down). 

The array of action `id`s is then passed to the state.

Some notes:

- `this.children_views` is where the currently sorted list items are stored. Not in the app. So my understanding is that this needs to be handled here, in the view.
- Looking at `selectedIndex`, `lastSelectedIndex`, and `arrayOfActionIdsToSelect` now, I should be able to use underscore methods for those.